lib/cogctl/actions/roles/info.ex

```diff
@@ -1,37 +1,36 @@
 defmodule Cogctl.Actions.Roles.Info do
   use Cogctl.Action, "roles info"
-  alias Cogctl.Table
+  import Cogctl.Actions.Roles.Util, only: [render: 4]
 
   def option_spec do
     [{:role, :undefined, :undefined, {:string, :undefined}, 'Role name (required)'}]
   end
 
   def run(options, _args, _config, endpoint) do
-    with_authentication(endpoint,
-                        &do_info(&1, :proplists.get_value(:role, options)))
-  end
-
-  defp do_info(_endpoint, :undefined) do
-    display_arguments_error
+    case convert_to_params(options, option_spec, [role: :required]) do
+      {:ok, params} ->
+        with_authentication(endpoint, &do_info(&1, params))
+      {:error, {:missing_params, missing_args}} ->
+        display_arguments_error(missing_args)
+    end
   end
 
-  defp do_info(endpoint, role_name) do
-    case CogApi.HTTP.Roles.show(endpoint, %{name: role_name}) do
+  defp do_info(endpoint, params) do
+    case CogApi.HTTP.Client.role_show(endpoint, %{name: params.role}) do
       {:ok, role} ->
-        role_attrs = [[Map.fetch!(role, :name), Map.fetch!(role, :id)]]
+        role_attr = [[Map.fetch!(role, :name), Map.fetch!(role, :id)]]
 
-        permission_attrs = Enum.map(role.permissions, fn(permission) ->
+        permission_attr = Enum.map(role.permissions, fn(permission) ->
                                [permission.namespace, permission.name, permission.id]
                            end)
-        display_output("""
-                       #{Table.format([["NAME", "ID"]] ++ role_attrs, false)}
 
-                       Permissions
-                       #{Table.format([["NAMESPACE", "NAME", "ID"]] ++ permission_attrs, true)}
-                       """ |> String.rstrip)
+        group_attr = Enum.map(role.groups, fn(group) ->
```

Comment:
It really feels like these sort of things should belong in the render method to me. I think it would be easier to see whats going on if you just passed your role to the render method and let it figure out how to render it. That way you aren't splitting up the rendering logic between two modules and this function, `do_info`, can just be responsible for pulling the correct data. It makes it easier to reason about what is going on here; and later, if we decide to change how we output something, we can just modify the module with the render function, instead of having to make changes in here and there. I think it will also make it easier to conditionally render bits based on user input, if we ever decide to do that.

So here we could just have something like:
```
...
{:ok, role} ->
  render(role)
...
```
Then over in the module that implements rendering:
```
...
def render(%Role{}=role) do
  #<implement the rendering bits here>
end
...
```